aframe-button-controls.js

```diff
@@ -128,10 +134,24 @@ AFRAME.registerComponent('button-controls', {
     addSessionEventListeners: function () {
         let sceneEl = this.el.sceneEl;
         if (!sceneEl.xrSession) { return; }
-        sceneEl.xrSession.addEventListener('selectstart', this.emitButton0DownEvent);
-        sceneEl.xrSession.addEventListener('selectend', this.emitButton0UpEvent);
-        sceneEl.xrSession.addEventListener('squeezestart', this.emitButton1DownEvent);
-        sceneEl.xrSession.addEventListener('squeezeend', this.emitButton1UpEvent);
+        if (this.data.poll) {
+            // normally with poll we don't use this method, but it's the only way to capture cardboard button presses.
+            sceneEl.xrSession.addEventListener('selectstart', evt => {
+                if (evt?.target?.inputSources[0]?.targetRayMode === "gaze") {
+                    this.pseudoCardboardController.buttons[0] = {pressed:true};
+                }
+            });
+            sceneEl.xrSession.addEventListener('selectend', evt => {
+                if (evt?.target?.inputSources[0]?.targetRayMode === "gaze") {
+                    this.pseudoCardboardController.buttons[0] = {pressed:false};
+                }
+            });
```

Comment:
Are squeezestart and squeezend not needed when poll is true?

Reply:
Cardboard is single-button, so there's only `select`. It seems to me that within the non-poll old-style API, you always listened for a grip and trigger button event only (which were `squeeze` and `select`). The gaze controller, however, exposes only one button, which fires as `select`.

Normally we wouldn't listen to _any_ events when `poll = true`, but in this hack to get the cardboard button working, we listen to those events for the `gaze` controller only, for this single button available only, and set the correct values to be observed by the `poll` loop.

Other values that would correspond to `squeeze` from other controllers are picked up in the 'normal' way, within the `poll` loop, when `poll = true`.